vault/rekey.go

```diff
@@ -98,15 +110,36 @@ func (c *Core) RekeyProgress(recovery bool) (int, error) {
 	return len(c.barrierRekeyProgress), nil
 }
 
+// RekeyVerifyProgress is used to return the rekey progress (num shares) during
+// verification.
+func (c *Core) RekeyVerifyProgress(recovery bool) (int, logical.HTTPCodedError) {
+	c.stateLock.RLock()
+	defer c.stateLock.RUnlock()
+	if c.sealed {
+		return 0, logical.CodedError(http.StatusServiceUnavailable, consts.ErrSealed.Error())
+	}
+	if c.standby {
+		return 0, logical.CodedError(http.StatusBadRequest, consts.ErrStandby.Error())
+	}
+
+	c.rekeyLock.RLock()
+	defer c.rekeyLock.RUnlock()
+
+	if recovery {
+		return len(c.recoveryRekeyVerifyProgress), nil
+	}
+	return len(c.barrierRekeyVerifyProgress), nil
+}
+
 // RekeyConfig is used to read the rekey configuration
-func (c *Core) RekeyConfig(recovery bool) (*SealConfig, error) {
+func (c *Core) RekeyConfig(recovery bool) (*SealConfig, logical.HTTPCodedError) {
```

Comment:
This error type change will also require some changes in `vault/rekey_test.go`, which no longer builds.

Reply:
Done!